### PR TITLE
test: Adds tests to the PublishedStatus component

### PR DESCRIPTION
**Pull request by michael-s-molina**
### SUMMARY
Adds tests to the `PublishedStatus` component.

### TEST PLAN
1 - Execute `PublishedStatus` tests
2 - All tests should pass

@rusackas @junlincc 

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/dashboard/components/PublishedStatus/PublishedStatus.test.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+      canSave
+      savePublished={savePublished}
+    />,
+  );
+  expect(screen.getByText('Published')).toBeInTheDocument();
+  userEvent.hover(screen.getByText('Published'));
+  expect(await screen.findByText(tooltip)).toBeInTheDocument();
+  expect(savePublished).not.toHaveBeenCalled();
+  userEvent.click(screen.getByText('Published'));
+  expect(savePublished).toHaveBeenCalledTimes(1);
```
Comment (yardz) [inline superset-frontend/src/dashboard/components/PublishedStatus/PublishedStatus.test.tsx:81]: Nice! (2)

File: superset-frontend/src/dashboard/components/PublishedStatus/PublishedStatus.test.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+      canSave
+      savePublished={savePublished}
+    />,
+  );
+  expect(screen.getByText('Draft')).toBeInTheDocument();
+  userEvent.hover(screen.getByText('Draft'));
+  expect(await screen.findByText(tooltip)).toBeInTheDocument();
+  expect(savePublished).not.toHaveBeenCalled();
+  userEvent.click(screen.getByText('Draft'));
+  expect(savePublished).toHaveBeenCalledTimes(1);
```
Comment (yardz) [inline superset-frontend/src/dashboard/components/PublishedStatus/PublishedStatus.test.tsx:56]: Nice!

**Review by yardz — APPROVED** [review]

**Review by geido — APPROVED** [review]
LGTM!

**Review by rusackas — APPROVED** [review]
LGTM!

**Pull request merged**